### PR TITLE
Add TextUI for SSH named hosts

### PR DESCRIPTION
**Pull request by smondet**
Cf. #212 

(and #370 was fixed on the way)

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/ketrew/371)

### CODE REVIEW

File: src/lib/named_hosts_text_ui.ml [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+                 s "Status: " % log_status status)))
+       @ normal);
+  return ()
+
+let display_details ~configuration id =
+  perform_call ~configuration (`Get_logs (id, `Full)) (function
+    | `Logs (id, markup) ->
+      Some markup
+    | oterh -> None)
+  >>= fun serialized_markup ->
```
Comment (rleonid) [inline src/lib/named_hosts_text_ui.ml]: `_other` ?

File: src/lib/named_hosts_text_ui.ml [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    | `Error err ->
+      fail (`Failure (fmt "Server answered with an ERROR: %s" err))
+    | other ->
+      begin match f down with
+      | Some s -> return s
+      | None ->
+      fail (`Failure (fmt "Unexpected process-holder down-message"))
+      end
+    end
+  end
```
Comment (rleonid) [inline src/lib/named_hosts_text_ui.ml]: The `begin`/`end` pairing in this function is confusing. The Ok/Error state is also confusing, specifically, why is `f` applied to `down` as opposed to `other` ?

Comment (rleonid) [inline src/lib/named_hosts_text_ui.ml]: Now I understand what it is doing, but still a bit awkward without understanding the context of the rest of the methods in this module.

**Comment by rleonid** [comment]
Looks fine in general

File: src/lib/client.ml [hunk 3]
```diff
@@ -185,6 +186,12 @@ module Http_client = struct
...
     call_up_msg (`Get_target_ids query)
 
+  let call_process_holder t message =
+    call_json t ~path:"/api" ~meta_meth:(`Post_message (`Process message))
+    >>= fun down ->
+    filter_down_message down ~loc:`Process_holder
+      ~f:(function `Process p -> Some p | other -> None)
+
 
 end
```
Comment (rleonid) [inline src/lib/client.ml]: Maybe add an `(* Http_client *)` to this `end` ?

File: src/lib/named_hosts_text_ui.ml (hunk 2 above)
Comment (smondet) [inline src/lib/named_hosts_text_ui.ml]: Yes, naming the message `` `Error of string `` makes it a bit confusing indeed. If one day we switch the 4.03.0 result type (regular variant), that will get fixed :)

File: src/lib/client.ml (hunk 3 above)
Comment (smondet) [inline src/lib/client.ml]: Not a fan of those comments; it's like encoding in the source something that should be a text-editor feature (like using `%` http://vim.wikia.com/wiki/Moving_to_matching_braces).

**Pull request merged**